ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -329,11 +329,9 @@ getCurrent LgrDB{..} = readTVar varDB
 setCurrent :: IOLike m => LgrDB m blk -> LedgerDB' blk -> STM m ()
 setCurrent LgrDB{..} = writeTVar $! varDB
 
-currentPoint :: forall blk. UpdateLedger blk => LedgerDB' blk -> Point blk
+currentPoint :: LedgerSupportsProtocol blk => LedgerDB' blk -> Point blk
 currentPoint = castPoint
-             . ledgerTipPoint (Proxy @blk)
-             . ledgerState
-             . LedgerDB.ledgerDbCurrent
+             . LedgerDB.ledgerDbTip
```

Comment:
Check again :suspect:

Reply:
It seems we're casting a point twice in a row. Is this needed? 

```haskell
ledgerDbTip :: IsLedger l => LedgerDB l -> Point l
ledgerDbTip = castPoint . getTip . ledgerDbCurrent
```

Reply:
Not sure where that excerpt comes from, I have the following:
```
-- | Reference to the block at the tip of the chain
ledgerDbTip :: IsLedger l => LedgerDB l -> Point (l EmptyMK)
ledgerDbTip = getTip . ledgerDbCurrent
```